p4_symbolic/symbolic/packet.cc

```diff
@@ -56,20 +55,59 @@ SymbolicPacket ExtractSymbolicPacket(SymbolicPerPacketState state) {
           GetOrDefault(state, "icmp.type", 8)};
 }
 
-ConcretePacket ExtractConcretePacket(SymbolicPacket packet, z3::model model) {
-  return {model.eval(packet.eth_src, true).to_string(),
-          model.eval(packet.eth_dst, true).to_string(),
-          model.eval(packet.eth_type, true).to_string(),
+gutil::StatusOr<ConcretePacket> ExtractConcretePacket(
+    SymbolicPacket packet, z3::model model,
+    const values::ValueFormatter &value_formatter) {
+  ASSIGN_OR_RETURN(
```

Comment:
Why does any of this change? IP addresses etc don't need translation. In fact, nothing in the packet ever needs translation.

Reply:
I can remove this. However, if in the future, some translation happens there, the tool will break. If no translation is needed, the function call is the identity.